zomes/test/src/lib.rs

```diff
@@ -195,6 +196,63 @@ fn emit_signal_from_sibling_cell(payload: SiblingEmitPayload) -> ExternResult<()
     Ok(())
 }
 
+#[hdk_extern]
+fn create_cap_grant_for_private_function(_: ()) -> ExternResult<CapSecret> {
+    let cap_secret = generate_cap_secret()?;
+
+    let mut functions: GrantedFunctions = BTreeSet::new();
+    functions.insert((zome_info()?.name, "private_function".into()));
+
+    create_cap_grant(CapGrantEntry {
+        tag: "".into(),
+        access: cap_secret.into(),
+        functions,
+    })?;
+
+    Ok(cap_secret)
+}
+
+#[hdk_extern]
+fn private_function(_: ()) -> ExternResult<TestObj> {
```

Comment:
@zo-el - I don't love the name `private_function` here, but couldn't find a name that more clearly communicated (without it being a terrible mouthful like `function_that_you_dont_have_automatic_access_to`) 😅

Lmk if you think of something better